test/core.plugin.tests.js

```diff
@@ -66,63 +63,282 @@ describe('Chart.plugins', function() {
+			expect(plugin.hook.calls.first().args[1]).toBe(42);
+			expect(plugin.hook.calls.first().args[2]).toEqual({});
 		});
 
-		it('should return TRUE if no plugin explicitly returns FALSE', function() {
-			Chart.plugins.register({
-				check: function() {}
+		it('should call global plugins with arguments', function() {
+			var plugin = {hook: function() {}};
+			var chart = window.acquireChart({});
+
+			spyOn(plugin, 'hook');
+
+			Chart.plugins.register(plugin);
+			Chart.plugins.notify(chart, 'hook', 42);
+			expect(plugin.hook.calls.count()).toBe(1);
+			expect(plugin.hook.calls.first().args[0]).toBe(chart);
+			expect(plugin.hook.calls.first().args[1]).toBe(42);
+			expect(plugin.hook.calls.first().args[2]).toEqual({});
+		});
+
+		it('should call plugin only once even if registered multiple times', function() {
+			var plugin = {hook: function() {}};
+			var chart = window.acquireChart({
+				plugins: [plugin, plugin]
 			});
-			Chart.plugins.register({
-				check: function() {
-					return;
-				}
+
+			spyOn(plugin, 'hook');
+
+			Chart.plugins.register([plugin, plugin]);
+			Chart.plugins.notify(chart, 'hook');
+			expect(plugin.hook.calls.count()).toBe(1);
+		});
+
+		it('should call plugins in the correct order (global first)', function() {
+			var results = [];
+			var chart = window.acquireChart({
+				plugins: [{
+					hook: function() {
+						results.push(1);
+					}
+				}, {
+					hook: function() {
+						results.push(2);
+					}
+				}, {
+					hook: function() {
+						results.push(3);
+					}
+				}]
 			});
-			Chart.plugins.register({
-				check: function() {
-					return null;
+
+			Chart.plugins.register([{
+				hook: function() {
+					results.push(4);
 				}
-			});
-			Chart.plugins.register({
-				check: function() {
-					return 42;
+			}, {
+				hook: function() {
+					results.push(5);
+				}
+			}, {
+				hook: function() {
+					results.push(6);
 				}
+			}]);
+
+			var ret = Chart.plugins.notify(chart, 'hook');
+			expect(ret).toBeTruthy();
+			expect(results).toEqual([4, 5, 6, 1, 2, 3]);
+		});
+
+		it('should return TRUE if no plugin explicitly returns FALSE', function() {
+			var chart = window.acquireChart({
+				plugins: [{
+					hook: function() {}
+				}, {
+					hook: function() {
+						return null;
+					}
+				}, {
+					hook: function() {
+						return 0;
+					}
+				}, {
+					hook: function() {
+						return true;
+					}
+				}, {
+					hook: function() {
+						return 1;
+					}
+				}]
+			});
+
+			var plugins = chart.config.plugins;
+			plugins.forEach(function(plugin) {
+				spyOn(plugin, 'hook').and.callThrough();
+			});
+
+			var ret = Chart.plugins.notify(chart, 'hook');
+			expect(ret).toBeTruthy();
+			plugins.forEach(function(plugin) {
+				expect(plugin.hook).toHaveBeenCalled();
 			});
-			var res = Chart.plugins.notify('check');
-			expect(res).toBeTruthy();
 		});
 
 		it('should return FALSE if no plugin explicitly returns FALSE', function() {
```

Comment:
This was already existing but I think it should be 'should return FALSE if any plugin explicitly returns FALSE'